### PR TITLE
Workaround for MySQL Bug #89139 (insertid reset after ping on 5.7 and 8.0)

### PR DESCRIPTION
**Pull request by dveeden**
https://bugs.mysql.com/bug.php?id=89139

Sending a ping resets the insert_id on 5.7.18, 5.7.20 and 8.0.3.
This doesn't happen with MySQL 5.6 and MariaDB 10.2.

Note that  https://bugs.mysql.com/bug.php?id=78778 is very similar but was fixed in 5.7.18.

```perl
#!/usr/bin/perl
use v5.24;
use DBI;

my $dbh = DBI->connect("DBI:mysql:database=test;host=127.0.0.1;port=5718", "msandbox", "msandbox");

my $sth = $dbh->prepare("INSERT INTO t1() VALUES()");
$sth->execute();

say "mysql_insertid: " . $sth->{'mysql_insertid'};
say "last_insert_id: " . $dbh->last_insert_id(undef, undef, undef, undef);

my $sth = $dbh->prepare("INSERT INTO t1() VALUES()");
$sth->execute();
$dbh->ping();

say "mysql_insertid: " . $sth->{'mysql_insertid'};
say "last_insert_id: " . $dbh->last_insert_id(undef, undef, undef, undef);
```

### CODE REVIEW

**Comment by pali** [comment]
You need to update also MANIFEST file when adding new (test) file (travis test was there fore it before big revert).

**Comment by dveeden** [comment]
The travis failure looks like it is not related to my commits.

**Comment by pali** [comment]
I'm saying that you need to put newly added (test) file into MANIFEST file and test for it is missing because due to that big revert.

File: t/33_last_insert_id.t [hunk 1]
```diff
@@ -0,0 +1,74 @@
+use strict;
```
Comment (pali) [inline t/33_last_insert_id.t]: What is the difference between `t/31insertid.t` and `t/33_last_insert_id.t` tests? They looks like are same.

Comment (dveeden) [reply t/33_last_insert_id.t]: One tests `$sth->{mysql_insertid}` and `$dbh->{mysql_insertid}` and the other tests `$dbh->last_insert_id(undef, undef, undef, undef)`

Comment (pali) [reply t/33_last_insert_id.t]: Then this should be in one test file. Because t/33_last_insert_id.t is big copy+paste.

**Comment by dveeden** [comment]
I did add the test file to the MANIFEST. However I didn't really understand the second part of your sentence.

**Comment by pali** [comment]
I included this patch into https://github.com/gooddata/DBD-MariaDB repository.

**Comment by CaptTofu** [comment]
thank you. Merged.

**Pull request merged**